### PR TITLE
feat: simplify parquet reader (#3282)

### PR DESCRIPTION
**Pull request by tustvold**
This PR migrates away from using ParquetExec to read files, in preparation for #3282.

Aside from drastically simplifying the code, this also fixes an issue where we were performing blocking IO, namely writing to a temporary file, in a regular tokio task. It also avoids having to use named temporary files, which avoids them accidentally getting orphaned in `/tmp` (see [here](https://docs.rs/tempfile/3.2.0/tempfile/#resource-leaking)).

Some very unscientific local testing showed no performance differential from this change (which isn't surprising)

### CODE REVIEW

File: parquet_file/src/storage.rs [hunk 1]
```diff
@@ -246,106 +192,44 @@ impl Storage {
...
+                break;
             }
         }
+
         Ok(())
     }
 
     pub fn read_filter(
-        predicate: &Predicate,
+        _predicate: &Predicate,
```
Comment (tustvold) [inline parquet_file/src/storage.rs:227]: This is temporary, but doesn't represent a regression as they weren't enabled before (#2163) and I don't think we are persisting parquet files with multiple row groups anyway. It should be relatively straightforward to re-enable in future should we decide to do so

File: parquet_file/src/storage.rs [hunk 2]
```diff
@@ -246,106 +192,46 @@ impl Storage {
...
-            .context(OpenTempFile)?;
-
-        debug!(?path, ?temp_file, "Beginning to read parquet to temp file");
-        let mut read_stream = store
-            .get_parquet_file(&path)
-            .await
+        let mut file = tempfile::tempfile().context(OpenTempFile)?;
+
+        debug!(?path, ?file, "Beginning to read parquet to temp file");
+        let read_stream = futures::executor::block_on(store.get_parquet_file(&path))
```
Comment (tustvold) [inline parquet_file/src/storage.rs:199]: This is doubly stupid in the case of a local filesystem, as not only are we copying the files to a temporary location from a location already on disk, but all the tokio filesystem IO is blocking and so we're needlessly proxying it via a threadpool only to use it in a blocking manner :laughing:

File: parquet_file/src/storage.rs (hunk 1 above)
Comment (alamb) [reply parquet_file/src/storage.rs:229]: See also https://github.com/influxdata/influxdb_iox/pull/2839 which did reenable it (but I haven't yet merged due to lack of test)

I am pretty sure I have seen iox make parquet files with multiple row groups, but I can't seem to find any at this moment. 

In any event, I think now that we are getting to the point where we can actually run queries on parquet files, we can figure out how, if at all, we want to re-enable pruning

**Review by alamb — APPROVED** [review]
This PR got me thinking that `ParquetExec` is the code that does Row Group Pruning, so that is something we would have to add to IOx in the future; If we could find some way to reuse the code in DataFusion (and the tests), that would be ideal. But we can deal with that as needed

**Comment by tustvold** [comment]
> If we could find some way to reuse the code in DataFusion (and the tests)

I _think_ this should be possible, after all we are already using `PruningPredicate` to prune chunks in IOx, and most of the mechanics of implementing row group pruning are actually implemented in `parquet` but I suppose we shall see.

**Pull request merged**